### ISSUE TITLE
Propose an easy way to provide dockerhub credentials

### ISSUE THREAD

**Issue by savmickael** [CS-SI/SafeScale#91]
Currently it is not possible to create a k8s cluster with a too large number of nodes due to [dockerhub pull rate limit](https://www.docker.com/increase-rate-limits).

The creation failed due to the fact that k8s need to pull calico image from dockerhub.

```
Warning  Failed   35m (x36 over 3h15m)     kubelet  Failed to pull image "calico/node:v3.10.4": rpc error: code = Unknown desc = Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Normal   Pulling  25m (x38 over 3h15m)     kubelet  Pulling image "calico/node:v3.10.4"
  Normal   BackOff  5m25s (x831 over 3h15m)  kubelet  Back-off pulling image "calico/node:v3.10.4"
  Warning  Failed   15s (x854 over 3h15m)    kubelet  Error: ImagePullBackOff
```

To overpass this dockerhub limitation, a solution is to buy a pro account on dockerhub. 

However it is currently not easy to pass these credentials (login and token) to k8s via safescale. You need to connect manually to the master to update k8s config with this credentials.

From my point of view it is a blocking issue to use safescale to create and manage large k8s cluster.

**Comment by vplanche-cssi** [comment]
Two things to come for this issue:

1. allow to define credentials for dockerhub in the feature docker.yml
2. allow to define alternate default registry in the feature docker.yml

The 2 points need to first enhance safescale CLI to be able to pass parameters to default features installed at cluster creation (docker, kubernetes, helm3, remotedesktop, ...), following this model:

`safescale cluster create -F k8s -C normal --parameter <feature_name>:<parameter_name>=<parameter_value> -p <parameter_name>=<parameter_value> ... <cluter_name>`

The first form `--parameter|-p <feature_name>:<parameter_name>` would concern the parameter of a specific feature, where the second one `--parameter|-p <parameter_name>:<parameter_value>` would concern all occurences of <parameter_name> in all features.

**Comment by oscarpicas** [comment]
1 Partially implemented in: 8d8902fe8744c7fa822fac1d3e3b452dfb2ac884 (develop branch)
-p DockerHubUsername:yourname
-p DockerHubPassword:yourpass

**Comment by oscarpicas** [comment]
Example of cluster creation using dockerhub:

safescale cluster create -F k8s -C Normal -parameter DockerHubUsername=youruser -p DockerHubPassword=yourpasswd mycluster
